doc/fwutil/fwutil.md

```diff
@@ -462,6 +540,200 @@ Aborted!
+`2.2.2.4.3 Platform Component Firmware Update Utility` explains the requirement of the component firmware upgrade utility
+to interfere with the fwutil to support the auto-update command - mainly status and install.
+
+**Note:**
+1. FW update will be disabled if component definition is not provided (e.g., 'BIOS': { })
+2. FW version will be read from image if `version` field is not provided
+
+##### 2.2.2.4.2 Description
+
+**The following command updates FW Automatically with installing the available component firmware, creating task(s) for update during reboot:**
+1. auto-update command for installation and reboot task creation
+```bash
+root@sonic:~# fwutil auto-update fw --yes --image=<current|next> --boot=<boot_type>
+Available FW updates with <boot_type> from <current|next> boot image  : COMPONENT1 (and COMPONENT2)
+<COMPONENT1> firmware update will be done during <boot_type> with version <#new_version> (current version #current_version)
+...
+<COMPONENT2> firmware update is starting with version <#new_version2> (current version #current_version2)
+...
+<COMPONENT2> firmware update is in progress with version <#new_version2> (current version #current_version2)
+...
+<COMPONENT2> firmware update is done during <boot_type> with version <#new_version> (current version #current_version)
+...
+All firmware update has been completed.
+```
+
+This is examples of a platform which supports the SSD firmware update during fast reboot and cold reboot with different commands
+along with CPLD firmware update available only for the cold reboot which can perform a power cycle triggered by some register setting.
+
+Example 1: SSD firmware update is available for fast-reboot upgrade path.
+SSD firmware update is logged in a platform defined designated file and it will be updated during fast-reboot.
+```bash
+root@sonic:~# fwutil auto-update fw --yes --image=next --boot=fast
+Available FW updates with fast-reboot from next boot image : SSD
+SSD firmware update will be done during fastreboot with version 5 (current version 4)
+All firmware update has been completed.
+```
+Example 2: SSD and CPLD update is available for cold reboot upgrade path.
+SSD firmware update is logged in a platform defined designated file and it will be updated during cold reboot.
+CPLD firmware update is going to be done with this command and the CPLD completion activity will be performed during cold reboot
+once all reboot processes are finished. The CPLD completion activity for this case is a power cycle triggered by the hw register setting.
+```bash
+root@sonic:~# fwutil auto-update fw --yes --image=next --boot=cold
+Available FW updates with fast-reboot from next boot image : SSD and CPLD
+SSD firmware update will be done during cold-reboot with version 5 (current version 4)
+...
+CPLD firmware update is in progress with version 5 (current version 4)
+...
+CPLD firmware update is completed with version 5 (current version 4)
```

Comment:
So, in the CPLD example above, the fwutil auto-update will be functionally equivalent to the existing "fwutil install" command, please mention this in the description.